### ISSUE TITLE
Properties "intended to be used with non-literal values"

### ISSUE THREAD

**Issue by tombaker** [dcmi/usage#27]
## 2018-06-11: PLEASE VOTE BELOW AT [COMMENT 294979379](https://github.com/dcmi/usage/issues/27#issuecomment-394979379)

See [ranges_intended.md](../blob/master/proposals/2018_iso-related/ranges_intended.md) - see also Issue #28 ("Definition of dct:relation")

The following properties

* [http://purl.org/dc/terms/hasFormat](http://dublincore.org/documents/dcmi-terms/#terms-hasFormat)
* [http://purl.org/dc/terms/hasPart](http://dublincore.org/documents/dcmi-terms/#terms-hasPart)
* [http://purl.org/dc/terms/hasVersion](http://dublincore.org/documents/dcmi-terms/#terms-hasVersion)
* [http://purl.org/dc/terms/isFormatOf](http://dublincore.org/documents/dcmi-terms/#terms-isFormatOf)
* [http://purl.org/dc/terms/isPartOf](http://dublincore.org/documents/dcmi-terms/#terms-isPartOf)
* [http://purl.org/dc/terms/isReferencedBy](http://dublincore.org/documents/dcmi-terms/#terms-isReferencedBy)
* [http://purl.org/dc/terms/isReplacedBy](http://dublincore.org/documents/dcmi-terms/#terms-isReplacedBy)
* [http://purl.org/dc/terms/isRequiredBy](http://dublincore.org/documents/dcmi-terms/#terms-isRequiredBy)
* [http://purl.org/dc/terms/isVersionOf](http://dublincore.org/documents/dcmi-terms/#terms-isVersionOf)
* [http://purl.org/dc/terms/references](http://dublincore.org/documents/dcmi-terms/#terms-references)
* [http://purl.org/dc/terms/relation](http://dublincore.org/documents/dcmi-terms/#terms-relation)
* [http://purl.org/dc/terms/replaces](http://dublincore.org/documents/dcmi-terms/#terms-replaces)
* [http://purl.org/dc/terms/requires](http://dublincore.org/documents/dcmi-terms/#terms-requires)
* [http://purl.org/dc/terms/source](http://dublincore.org/documents/dcmi-terms/#terms-source)
* [http://purl.org/dc/terms/subject](http://dublincore.org/documents/dcmi-terms/#terms-subject)

have the following comment:

    This term is intended to be used with non-literal values as defined in the
    DCMI Abstract Model.  As of December 2007, the DCMI Usage Board is seeking
    a way to express this intention with a formal range declaration.

**Comment by tombaker** [comment]
One option is to declare them to be of type http://www.w3.org/2002/07/owl#ObjectProperty - however would we not then have to do this across the board, for all properties with non-literal ranges?  And do we _really_ want to do this?

**Comment by osma** [comment]
This is related to the same underlying problem as issue #32 and the solution could be similar - use a `rangeIncludes` property to indicate the recommended or expected range.

**Comment by tombaker** [comment]
@osma So what would the expected range be? Except for `dct:subject`, which could have a range of `skos:Concept`, all of the properties relate one (generic) resource to another (generic) resource.  But would [`schema:rangeIncludes rdfs:Resource`](https://www.w3.org/TR/rdf-schema/#ch_resource) really _say_ anything if the range of every property can already be inferred to be `rdfs:Resource` (and of which `rdfs:Literal` is a subclass)?  Only half-seriously... would we want to coin a term `dct:Resource`?  How would it be defined?

**Comment by osma** [comment]
Funny, I wasn't aware that `rdfs:Literal` is a subclass of `rdfs:Resource`. You learn something new every day!

DC properties have always been ambiguous on the domain side, since they can be used with resources of any type. Coining a new class `dct:Resource` and using it for both domains and ranges would be one option for sure, though I'm not sure it would be very useful for anyone.

Maybe what I said above wasn't such a great idea. I only thought about it from the perspective of assigning a "soft" range. But the problem here is really that for most of these properties there is no good candidate for what class that range should be.

Perhaps we need a "soft" version of `owl:ObjectProperty`, something like `dcam:PropertyIntendedToBeUsedWithNonLiteralValues` :)

**Comment by tombaker** [comment]
Half-seriously... maybe we need to update the statement to read:

    This term is intended to be used with non-literal values.
    As of June 2018, the DCMI Usage Board is seeking
    a way to express this intention with a formal range declaration.

Slightly more seriously, could we just add a usage comment:

    This term is intended to be used with non-literal values.

instead of trying to state it more formally, or in ways that everyone would likely ignore?

**Comment by stuartasutton** [comment]
In your half-serious vein, that last time we made such a comment (can't remember exactly where*) the "seeking" (or at least the comment) was around for years...resulting in a seeming dereliction of duty, or frightfully slow thinking.

**Comment by tombaker** [comment]
What are the arguments for and against saying they are `owl:ObjectProperty`?

**Comment by osma** [comment]
Defining them as `owl:ObjectProperty` would conflict with actual usage where literal values are often used. Of course using literal values goes against the current recommendation too, but this would make it more formal and thus more strict.

Not sure whether this would be a problem in practice. But if I had to design an AP and knew I would have only literal values for e.g. source, I would avoid using `dct:source` if it were an ObjectProperty, and instead either coin my own property or use the old `dc:source` from DC Elements 1.1. So one could argue that making these ObjectProperties would make them less likely to be reused. I'd rather have properties that work for many use cases.

**Comment by tombaker** [comment]
@osma
> Defining them as `owl:ObjectProperty` would conflict with actual usage where literal values are often used. Of course using literal values goes against the current recommendation too, but this would make it more formal and thus more strict.
> 
> Not sure whether this would be a problem in practice. But if I had to design an AP and knew I would have only literal values for e.g. source, I would avoid using `dct:source` if it were an ObjectProperty, and instead either coin my own property or use the old `dc:source` from DC Elements 1.1. So one could argue that making these ObjectProperties would make them less likely to be reused. I'd rather have properties that work for many use cases.

Very well put!  I support this position.  Put another way, it means deliberately erring on the side of semantic undercommitment.  Do you agree we should revise the usage comment to continue to say the following?

    This property is intended to be used with non-literal values.

**Comment by tombaker** [comment]
If we were to:
* convert all `rdfs:range` statements into `dcam:rangeIncludes` (and ditto for domain)
* deprecate the "lumpy" range classes 'LocationPeriodOrJurisdiction' and 'MediaTypeOrExtent' in favor of listing multiple expected ranges
  * See: http://dublincore.org/documents/2008/01/14/domain-range/classes.jpg

Then I see the following consequences for the editor of the ISO 15836-2 draft:
* add definitions for "expected range", etc, in section 3.2
* change all "Range" and "Domain" term attributes to "Expected Range" and "Expected Domain"
* dropping the "lumpy" classes
* edit entries that use the "lumpy" classes

In short, the impact on ISO 15836-2 seen strictly from an _editorial_ point of view would be quite minimal.

A decision along these lines would make it easier (should we decide to do so) to correct some other awkward [range issues](https://github.com/dcmi/usage/labels/ranges), such as Issue #22 (Language).

**Comment by tombaker** [comment]
@kcoyle
> Are there other DCAM terms that are defined? This seems like a leap to start adding a DCAM namespace to DC Terms when, in fact, the trend seems to be to abandon DCAM altogether as not having gained any traction over 10+ years. 

There is in fact already a DCAM namespace in DCMI Terms - see http://dublincore.org/documents/dcmi-terms/#section-8 :

* the class VocabularyEncodingScheme (which should arguably be equated now with SKOS Concept Scheme)
* the property memberOf ("A relationship between a resource and a vocabulary encoding scheme which indicates that the resource is a member of a set.")

Admittedly these have been pretty much obsoleted by SKOS, but the namespace is there to be used, and it is even mentioned in the [DCMI Namespace Policy](http://dublincore.org/documents/dcmi-namespace/), Section 2, so we would not need to start a new namespace (and add it to the Namespace Policy, etc) to accommodate `[domain|range]Includes` properties.

We can always rename Section 2 of the DC Terms document as long as we can live with the fact that the namespace URI contains the string 'd-c-a-m', which in this case seems like a small price to pay.

**Comment by tombaker** [comment]
Hmm - "email replies do not support Markdown"...

**Comment by kcoyle** [comment]
Tom, using DCAM as a namespace would make sense if we were defining the DCAM ontology terms, but AFAIK this is unrelated to DCAM. And, as I said above, DCAM has been pretty much self-deprecated due to lack of uptake. Calling this DCAM does not seem to me to be a good idea.

**Comment by danbri** [comment]
I'm not sure what is best to do for this case, beyond at least removing the bit about 2007 :)

**Comment by kcoyle** [comment]
How about simply:

This term is intended to be used with non-literal values

I don't think there are more than a handful of people on the planet who would know what  non-literal "as defined in the DCMI Abstract Model" means, but there _are_ people who know what "literal" means in RDF, and could possibly parse "non-" in relation to that.

**Comment by kaiec** [comment]
Either this or using dcam:rangeIncludes to point to meaingful classes that we know are used or that we recommend.

**Comment by tombaker** [comment]
**PROPOSAL FOR STRAW VOTE**
Change

    This term is intended to be used with non-literal values as defined in the
    DCMI Abstract Model.  As of December 2007, the DCMI Usage Board is seeking
    a way to express this intention with a formal range declaration.

to read, simply:

    This property is intended to be used with non-literal values.

for all properties listed in https://github.com/dcmi/usage/issues/27#issue-328829121 except for dct:subject.

Discussion:
* Uses "property" instead of the vaguer "term".
* May require that we define "non-literal value" (and indeed "value") somewhere.
* Drops the notion that the UB is "seeking a way to express this intention with a formal range declaration".
* dct:subject, the exception, could have a range (or rangeIncludes) of skos:Concept

Please vote by adding thumbs-up/down to this posting.

**Comment by tombaker** [comment]
@kcoyle
> I don't think there are more than a handful of people on the planet
> who would know what non-literal "as defined in the DCMI Abstract
> Model" means, but there _are_ people who know what "literal" means in
> RDF, and could possibly parse "non-" in relation to that. 

It is an interesting question whether we need to define "non-literal
value" or indeed "value", which is used several times in the ISO 15836-2
draft but never defined.  Perhaps, as you hint, we do not need to
define these terms at all?

If we did want to define them, I see two places in ISO 15836-2 where one
might do so:

1. Section 3.1 (Terms and definitions), p. 9 - see 
   https://github.com/dcmi/iso15836-2/blob/master/drafts/ISO_CD_15836-2_2018-06-04.pdf
   (protected). (FWIW, the DCMI Abstract Model defined it as: "A value
   which is a physical, digital, or conceptual entity." where "value" was
   defined as "The physical entity, conceptual entity or literal (a
   resource) that is associated with a property when a property-value pair
   is used to describe a resource.")  Section 3.1.3 already defines "literal" 
   as "string of Unicode characters, typically letters or integers".

2. In the non-normative Annex B ("Dublin Core metadata as linked data"),
   the text of which I have posted at 
   https://github.com/dcmi/usage/blob/master/proposals/2018_iso-related/annexb_linkeddata.md .

@kcoyle would you like to propose a definition for "non-literal value"
(and possibly "value")?

**Comment by osma** [comment]
If we're talking RDF, then the world of values consists of literals, URI references and blank nodes - nothing else. Thus non-literal means either URIref or bnode.

**Comment by kcoyle** [comment]
Would "dct:subject, the exception, could have a range (or rangeIncludes) of skos:Concept" require that the value has been defined elsewhere as a skos:Concept? If that is the case, then I would vote no on that because it would eliminate any non-SKOS vocabularies (of which many remain today). Also, there are surely subject lists in use that are strings, such as those that use keywords taken from abstracts. So dct:subject's range, if anything, should be IRI or literal, which means that the range can be anything, no?

**Comment by osma** [comment]
@kcoyle If we would set the `rdfs:range` for `dct:subject` as `skos:Concept`, then that would indeed cause problems for non-SKOS subjects, including literal keywords. But lately the talk has been about using `rangeIncludes`, which is more of a recommendation than a strict rule and doesn't exclude using something else than a skos:Concept as the value.

I'm still unsure about whether `skos:Concept` makes sense even as a `rangeIncludes` value for `dct:subject`. In bibliographic data I've published at NatLibFi, I've used the similar property [schema:about](http://schema.org/about) with different kinds of values including `skos:Concept`, `schema:Person`, `schema:Organization` and `schema:CreativeWork` instances, probably others too. Schema.org defines the expected type (`rangeIncludes`) of `schema:about` as `schema:Thing` which is pretty broad and although it seems to exclude literal values, the [Getting Started](http://schema.org/docs/gs.html#schemaorg_expected) docs make it clear that even literal values are OK.

**Comment by kcoyle** [comment]
It seems to me that we achieve our goal by keeping the note "This term is intended to be used with non-literal values" and omitting range altogether. I say this because there appear to be a number of possible expected values and none appear to predominate.

**Comment by makxdekkers** [comment]
@tombaker Two points about the exception for dct:subject:

1. why would it not be valid for someone to use the text of a Libary of Congress subject heading for subject, or just "Politics"? If the proposal is for other properties to make it easy for publishers, what is the argument to do it differently for dct:subject?
2. specifying a range of skos:Concept for dct:subject is a narrowing of semantics. With the current definition, anything can be the subject of a resource, including  people, places etc. Narrowing it may make existing implementations non-conformant.

**Comment by osma** [comment]
I voted +1 in the straw poll above because the general idea makes sense; however, I think we should do the same for `dct:subject` too, just like for the other properties.

**Comment by kaiec** [comment]
Based on the other discussion I think rdfs:range is basically off the table, mainly, however, because it conflicts with literal values. I wouldn't mind that much about other possible types because in my opinion the used objects do not need to be typed explicitly as skos:Concpts, the range would simply say that whatever is used can be regarded as a skos:Conspet, which seems to be true if it can be used as a subject of something.

Anyhow, I think this is all solved perfectly with the rangeIncludes property which I start to like very much. And if we have such a way to actually provide guidance on possible or recommended types to be used for objects in a formal way, I am for it. Of course then it has to be discussed which types deserve such a formal endorsement, but skos:Concept seems to be a quite obvious canddate for dct:subject.

**Comment by tombaker** [comment]
@makxdekkers While `dct:subject rdfs:range skos:Concept` would not fit with a literal value, it would not conflict with having a person or place as the value because in a formal sense, there is nothing that cannot also be a SKOS concept (except for a [SKOS concept scheme](https://www.w3.org/TR/skos-reference/#L1228) or [collection](https://www.w3.org/TR/skos-reference/#L3424)).  The association would be even weaker using a form of `rangeIncludes`.  As I see it, saying `dct:subject rangeIncludes skos:Concept` would essentially just be a slightly more formal way to say "this property is intended to be used with non-literal values", and with `rangeIncludes` it would not even preclude the literal value.

**Comment by makxdekkers** [comment]
@kaiec I am just wondering why you would want to restrict the range to skos:Concept. What benefit does it bring? Isn't this saying that literlly anything is/can be a skos:Concept? Does that give you any inforamtion at all?

**Comment by tombaker** [comment]
@kcoyle
> Would "dct:subject, the exception, could have a range (or rangeIncludes) of skos:Concept" require that the value has been defined elsewhere as a skos:Concept?

As I see it, the issue is not whether something was defined somewhere as a `skos:Concept`, because anything can safely (i.e., without contradiction) be _inferred_ to be a `skos:Concept` (with two exceptions as above), but whether we want to take one small step beyond simply saying in words that it is intended to be used with a non-literal value, to making that same statement in a slightly more formal way by saying `rangeIncludes skos:Concept`.  To use [Makx's metaphor](https://github.com/dcmi/usage/issues/32#issuecomment-394735551), the question is whether a stop sign will do or whether a traffic light would help.

**Comment by tombaker** [comment]
@osma 
> I'm still unsure about whether skos:Concept makes sense even as a rangeIncludes value for dct:subject. 

I meant it as a "soft" version of `owl:ObjectProperty` in the spirit of your [suggestion](https://github.com/dcmi/usage/issues/27#issuecomment-394334769):

> Perhaps we need a "soft" version of owl:ObjectProperty, something like dcam:PropertyIntendedToBeUsedWithNonLiteralValues :)

But while `skos:Concept` may clear the very low bar of qualifying as a `rangeIncludes` value, I admit that this is perhaps not reason enough to do it this way.

**Comment by tombaker** [comment]
See [ranges_intended.md](../blob/master/proposals/2018_iso-related/ranges_intended.md) - see also Issue #28 ("Definition of dct:relation")

Please vote by adding thumbs-up/down to this posting.

----------------------------------------------------------------------
**PROPOSED FOR FORMAL VOTE**

Change the existing comment:

    This term is intended to be used with non-literal values as defined in the
    DCMI Abstract Model.  As of December 2007, the DCMI Usage Board is seeking
    a way to express this intention with a formal range declaration.

to read:

    This property is intended to be used with non-literal values.

for the following properties:

* [http://purl.org/dc/terms/hasFormat](http://dublincore.org/documents/dcmi-terms/#terms-hasFormat)
* [http://purl.org/dc/terms/hasPart](http://dublincore.org/documents/dcmi-terms/#terms-hasPart)
* [http://purl.org/dc/terms/hasVersion](http://dublincore.org/documents/dcmi-terms/#terms-hasVersion)
* [http://purl.org/dc/terms/isFormatOf](http://dublincore.org/documents/dcmi-terms/#terms-isFormatOf)
* [http://purl.org/dc/terms/isPartOf](http://dublincore.org/documents/dcmi-terms/#terms-isPartOf)
* [http://purl.org/dc/terms/isReferencedBy](http://dublincore.org/documents/dcmi-terms/#terms-isReferencedBy)
* [http://purl.org/dc/terms/isReplacedBy](http://dublincore.org/documents/dcmi-terms/#terms-isReplacedBy)
* [http://purl.org/dc/terms/isRequiredBy](http://dublincore.org/documents/dcmi-terms/#terms-isRequiredBy)
* [http://purl.org/dc/terms/isVersionOf](http://dublincore.org/documents/dcmi-terms/#terms-isVersionOf)
* [http://purl.org/dc/terms/references](http://dublincore.org/documents/dcmi-terms/#terms-references)
* [http://purl.org/dc/terms/relation](http://dublincore.org/documents/dcmi-terms/#terms-relation)
* [http://purl.org/dc/terms/replaces](http://dublincore.org/documents/dcmi-terms/#terms-replaces)
* [http://purl.org/dc/terms/requires](http://dublincore.org/documents/dcmi-terms/#terms-requires)
* [http://purl.org/dc/terms/source](http://dublincore.org/documents/dcmi-terms/#terms-source)
* [http://purl.org/dc/terms/subject](http://dublincore.org/documents/dcmi-terms/#terms-subject)

Discussion

This formulation drops the notion that the Usage Board is (still) seeking a way
to express this intention with a formal range declaration while retaining the message
that the property is intended to be used with non-literal values.

Defining these properties as instances `owl:ObjectProperty` would express the
same intention more formally and strictly, though at the cost of more strongly
and explicitly conflicting with actual usage where literal values are often
used.

Open questions

* Should "non-literal value" (and "value") be defined somewhere?

**Comment by jneubert** [comment]
For dct:subject, perhaps adding an example could clarify that anything may be a subject:
```
EXAMPLE: http://viaf.org/viaf/110981647
         http://vocab.getty.edu/aat/300033618
         (a book about Frieda Kahlo and her paintings is indexed by the URI
         for "Frieda Kahlo" in VIAF and the URI for "paintings" in AAT thesaurus)
  
```
A class skos:Concept could be inferred from an according range, but (on second thougt) I don't think we should require the the users of the spec to take that semweb-ish hurdle.

**Comment by makxdekkers** [comment]
Agree with @jneubert

**Comment by tombaker** [comment]
@jneubert Note that I have withdrawn the proposal and list `dct:subject` as one of the properties that should simply have the comment about being "intended to be used with non-literal values".  Whether we should, in addition, provide examples (as you suggest) is, I think, a separate issue that we can turn into a formal proposal later.

**Comment by jneubert** [comment]
@tombaker OK, fine with me

**Comment by kcoyle** [comment]
I would back-burner the question about defining value and non-literal at the moment, but keep a running list of terms that may need definitions. Both of these are very commonly used in IT.

**Comment by aisaac** [comment]
@kcoyle @tombaker for the record I can see some interest in trying to define "non-literal", especially as in that process we may find a better formulation for the resolution to this issue ;-). But I would be in favour of trying not to define 'value' because (1) it's probably going to be a can of worms (2) I think it's actually nice to keep some vagueness, which could keep DC compatible with a variety of (non-RDF) data models and formats, and replacing 'value' by something more precise could make us lose this flexibility.

**Comment by tombaker** [comment]
With 9 votes for and 0 against, [ranges_intended.md](../blob/master/proposals/2018_iso-related/ranges_intended.md) is APPROVED.

Note: It might at some point be good to define "non-literal".

**Comment by juhahakala** [comment]
The ISO draft uses now "This property is intended to be used with non-literal values" instead of the old, longer text.
